velociraptor/tools/lines.py

```diff
@@ -83,13 +100,25 @@ def binned_mean_line(
 
             centers.append(0.5 * (x_bins[bin - 1].value + x_bins[bin].value))
 
+        # If the number of data points in the bin is less than minimum_in_bin,
+        # collect these data points if needed
+        elif (minimum_in_bin > indicies_in_this_bin.sum() > 0) and return_additional:
```

Comment:
Don't need additional check against `minimum_in_bin` and you should cache the result of `indicies_in_this_bin.sum()` at the top. So this line would be:

```
elif number_in_bin > 0 and return_additional:
```

Reply:
Indeed, this will be much better!